dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp

```diff
@@ -468,3 +472,62 @@ void CommandBarFlyoutCommandBar::UpdateTemplateSettings()
         }
     }
 }
+
+void CommandBarFlyoutCommandBar::UpdateShadow()
+{
+    if (PrimaryCommands().Size() > 0)
+    {
+        AddShadow();
+    }
+    else if (PrimaryCommands().Size() == 0)
+    {
+        ClearShadow();
+    }
+}
+
+void CommandBarFlyoutCommandBar::AddShadow()
+{
+    if (SharedHelpers::IsThemeShadowAvailable())
+    {
+#ifdef USE_INSIDER_SDK
+        //Apply Shadow on the Grid named "ContentRoot", this is the first element below
+        //the clip animation of the commandBar. This guarantees that shadow respects the 
+        //animation
+        winrt::IControlProtected thisAsControlProtected = *this;
+        auto grid = GetTemplateChildT<winrt::Grid>(L"ContentRoot", thisAsControlProtected);
+        if (winrt::IUIElement10 grid_uiElement10 = grid)
+        {
+            if (grid_uiElement10.Shadow() == nullptr)
+            {
+                winrt::Windows::UI::Xaml::Media::ThemeShadow shadow;
+                grid_uiElement10.Shadow(shadow);
+
+                auto translation = winrt::float3{ grid.Translation().x, grid.Translation().y, 32.0f };
+                grid.Translation(translation);
+            }
+        }
+#endif
+    }
+}
+
+void CommandBarFlyoutCommandBar::ClearShadow()
+{
+    if (SharedHelpers::IsThemeShadowAvailable())
+    {
+#ifdef USE_INSIDER_SDK
+        winrt::IControlProtected thisAsControlProtected = *this;
+        auto grid = GetTemplateChildT<winrt::Grid>(L"ContentRoot", thisAsControlProtected);
+        if (winrt::IUIElement10 grid_uiElement10 = grid)
+        {
+            if (grid_uiElement10.Shadow() != nullptr)
```

Comment:
Don't test against nullptr.

```suggestion
            if (grid_uiElement10.Shadow())
```